### PR TITLE
Output as generator with lazy-loading pagination

### PR DESCRIPTION
**Pull request by kevhill**
This is a backward incompatible change to the way you use the pipedrive client, but I found it useful so I'm putting it up here for discussion.

Basically, I ended up writing a bunch of management code to get more than 100 things to handle pipedrive's pagination. Instead I really just wanted to write simple `for` loops over the results. So, I decided to change the wrapper into something that returned a python-style generator. The interface is quite nice for basic things, but if you rely on using `response['additional_data']` then that is now obscured.

tldr, you can now do this:
```python
pd = Pipedrive(PIPEDRIVE_API_TOKEN)

for person in pd.persons()
    # each person object is a dict representing the data on that person
    func(person)
```

### CODE REVIEW

File: pipedrive/__init__.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def _request(self, endpoint, data, method='POST'):
             response, data = self.http.request(uri, method=method, headers={'Content-Type': 'application/x-www-form-urlencoded'})
         else:
             uri = PIPEDRIVE_API_URL + endpoint + '?api_token=' + str(self.api_token)
-            response, data = self.http.request(uri, method=method, body=urlencode(data), headers={'Content-Type': 'application/x-www-form-urlencoded'})
+            response, data = self.http.request(uri, method=method, body=json.dumps(data), headers={'Content-Type': 'application/json'})
```
Comment (kevhill) [inline pipedrive/__init__.py]: "We recommend using JSON body format when performing API requests" https://developers.pipedrive.com/docs/api/v1/#/

**Comment by jscott1989** [comment]
Looks good to me - but I don't use the library any more so I'll wait and see if there's more feedback on it either way okay? :)

**Comment by Dunedan** [comment]
Using it with this PR and it works fine. :+1:

File: pipedrive/__init__.py (hunk 1 above)
Comment (bieli) [reply pipedrive/__init__.py:38]: We can see general issues with Pipedrive API server implementation (wrong specification of API server - they rely on "Content-Type" header instead of "Accept" - look to RFC: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept):
1) When using `requests` Python library, I can see HTML instead of JSON the same with curl
2) Documentation doesn't follow HTTP RFC in case of headers request specification, what is wrong today (maybe 20 years ago it would have been fine, but not in 2024!) -> https://pipedrive.readme.io/docs/core-api-concepts-requests

Sometimes junior developers needs to listen critical comments carefully :-) 
This is not comment to you @kevhill - this is for Pipedrive API server developers :-)
